### DISCUSSION TITLE
Don't forward traffic through the tunnel

### DISCUSSION THREAD

**Discussion by vga-diver** [XTLS/Xray-core#3047] [General]
Hi there,
Can somebody please advise is it possible in Xray avoid forwarding traffic through the X-ray tunnel but use instead clients local gateway.

Example:
I have a list of dns names/ips to which I wan't to use client's gateway (don't know it's ip, it could be different)  and don't want to route this traffic trough the X-ray server.

Or probably, to have default system route as is and for a particular dns names route traffic to the x-ray "freedom"

**Comment by yomnxkcs** [comment]
Maybe you can take a look at this project: https://github.com/daeuniverse/dae

**Reply by vga-diver** [reply]
thanks, @yomnxkcs 
But that project looks like proxy solution. 
However in my case I need a VPN with the ability to split traffic and decide whether it(traffic) should go to the internet/private subnet through the X-ray server or directly to the internet without tunnel.

**Comment by x90nik** [comment]
https://www.proxifier.com/
https://www.yogadns.com/

**Reply by vga-diver** [reply]
hi @mrgharabaghi , thanks , but these are also more proxies rather than vpn solutions

**Comment by pulsarice** [comment]
Why don't you use xray's internal routing rules? you can define list of IPs and Domains and route them to Freedom (direct) outbound which bypasses the proxy connection and uses your default gateway.
Which client are you using? many like v2ray have the ability to easily configure routing rules via user interface.

**Reply by vga-diver** [reply]
Hi @pulsarice,
thanks for the advise, that was my idea, but I didn't find how that outbound can be configured.
By default, I have the following outbounds:
```
  "outbounds": [
    {
      "tag": "direct",
      "protocol": "freedom",
      "settings": {}
    },
    {
      "tag": "blocked",
      "protocol": "blackhole",
      "settings": {}
    }
  ],
```
As I noticed that freedom sends traffic using X-Ray's default route rather than client's route...
e.g. this route will send traffic for test-user to google.com through the vpn server
```
      {
        "type": "field",
        "outboundTag": "direct",
        "domain": [
          "google.com"
        ],
        "user": [
          "test-user"
        ]
      },
```

**Reply by pulsarice** [reply]
Is this your client configuration or server's? as I don't see any proxy outbound.
The routing configuration should be done on user's side. you can't control client's behaviour from server side.

**Reply by vga-diver** [reply]
thanks!
Right, that was a part of my server's config.
So, X-ray doesn't allow to populate client's routes from server side, right? and there is no option in configs for clients' routes?

e.g. I provide my client only a qr code without any additional information. it is generated on the server side and in the client configuration section there is no any additional fields for the client's routes or any other non-default configs...

**Reply by pulsarice** [reply]
>So, X-ray doesn't allow to populate client's routes from server side, right?

That's my understanding.

some client's (like foxray) have the ability to get routing rules via an additional link. but others should be configured manually.
[This is a Persian guide](https://telegra.ph/%D9%86%D8%AD%D9%88%D9%87-%D9%85%D8%B3%D8%AF%D9%88%D8%AF%D8%B3%D8%A7%D8%B2%DB%8C-%D8%B3%D8%A7%DB%8C%D8%AA%E2%80%8C%D9%87%D8%A7-%D9%88-%D8%A7%D9%BE%E2%80%8C%E2%80%8C%D9%87%D8%A7%DB%8C-%D8%A7%DB%8C%D8%B1%D8%A7%D9%86%DB%8C-%D8%A7%D8%B2-%D8%B3%D9%85%D8%AA-%DA%A9%D8%A7%D8%B1%D8%A8%D8%B1-02-15) for configuration of routing rules in different clients apps.
You can make a similar guide for your users to configure their clients accordingly. 
I also did something on my own server, I blocked the destinations that users should not use the proxy for, so users now have to configure their clients, otherwise they won't be able to connect to those destinations.
